File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/memento/memento/Originator.java
package com.bigboxcode.designpattern.memento.memento;

public class Originator {

    private String state;

    public void setState(String state) {
        this.state = state;
    }

    public String getState() {
        return state;
    }

    public Memento setMemento() {
        System.out.println("Memento Saved with timestamp => " + state);

        return new Memento(state);
    }

    public void getMementoState(Memento memento) {
        state = memento.getState();
    }
}

File: design-patterns/php/memento/general/Originator.php
<?php
// Originator.php

namespace BigBoxCode\DesignPattern\Memento\General;

class Originator {
    private string $state = '';

    public function setState(string $state): void {
        $this->state = $state;
    }

    public function getState(): string {
        return $this->state;
    }

    public function setMemento(): Memento {
        echo "Memento Saved with timestamp => " . $this->state . "\n";

        return new Memento($this->state);
    }

    public function getMementoState(Memento $memento): void {
        $this->state = $memento->getState();
    }
}